src/draw/text.rs

```diff
@@ -129,6 +129,7 @@ impl Draw for TextState {
                 }
             }
         }
+        renderer.builder.push_rect(&PrimitiveInfo::new(bounds.clone()), self.background_color.into());
```

Comment:
You might as well still check if background color is transparent before adding the rect as suggested above, it's just a small optimization but there's no reason not to. Then squash the commits and I'll merge